openedx/features/content_type_gating/services.py

```diff
@@ -29,4 +46,40 @@ def content_type_gate_for_block(self, user, block, course_id):
             access = has_access(user, 'load', block, course_id)
             if (not access and access.error_code == 'incorrect_user_group'):
                 return access.user_fragment
+
+            # Handle case of staff users masquerading as Learner in Audit or Learner in Limited Access
+            if self._is_masquerading_as_audit_or_limited_access(user, course_id):
+                # Check merged_group_access in case the course steam overrode content type gating
+                # for this problem with the studio override
+                if LIMITED_ACCESS.id not in block.merged_group_access.get(CONTENT_GATING_PARTITION_ID):
```

Comment:
Should this check be brought up and out of the audit_or_limited block? Is it actually connected to the audit/limited masquerade path or just something we should always look at? Like, if we later add another toplevel check for some other condition, we don't want to have to redo this check.

Maybe something at the toplevel like:
```
if LIMITED_ACCESS.id in block.merged_group_access.get(CONTENT_GATING_PARTITION_ID):
    return None

if audit_or_limited:
    ...
```

But I might also be misunderstanding this merged group access bit. Is `has_access` is already making this check? In which case, we only need to re-check it if we're doing something vague like the masquerading... If so, the way you had it seems good.

But that makes me wonder... should we be doing this masquerade check inside `has_access` instead of this method?

Reply:
has_access handles all this by default, but it didn't seem to be working when masquerading as one of the special groups
perhaps i should be trying to modify has_access, not sure if that would have any dangerous side_effects. 
perhaps @cpennington has thoughts?

Reply:
@mikix turns out I was able to just delete this portion.  
In my initial bugfix I was reusing the _get_user method that retrieved the user through
get_user_model(). 
when I switched over to using request.user it turns out the masquerade information that got passed through made has_access work the way we want in this method